### ISSUE TITLE
uri module executed twice

### ISSUE THREAD

**Issue by lmilbaum** [ansible/ansible#32978]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
uri

##### ANSIBLE VERSION
```
ansible 2.4.1.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]

```

##### OS / ENVIRONMENT
control machine - docker container Ubuntu 16.04

##### SUMMARY

Rest API command executed via uri module is triggered twice although specified in the playbook only once

##### STEPS TO REPRODUCE

```yaml

- name: MedOne | deploy server
  uri:
    method: POST
    url: "https://{{ iaas.cloud_api_url }}/caas/2.6/{{ iaas.org_id }}/server/deployServer"
    body: "{{ lookup('file','/var/tmp/server.json') }}"
    body_format: "json"
    user: "{{ iaas.user_name }}"
    password: "{{ iaas.password }}"
    force_basic_auth: yes
    return_content: yes
    headers:
      Content-Type: "application/json"
   register: deploy_server_response

```

##### EXPECTED RESULTS
The rest API command executed once

##### ACTUAL RESULTS
The rest API command executed twice

**Comment by ansibot** [comment]
cc @romeotheriault
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Are you saying that the task was executed by ansible 2 times, as seen in the output of ansible-playbook?  Or are you saying that the API was hit twice in context of one task execution?

I'm going to assume the 2nd one.

With that assumption, was it a POST request both times?  Can you provide more information?  Did you see this in logs? Can you provide those logs?

needs_info

**Comment by ansibot** [comment]
@lioramilbaum This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by biz812** [comment]
I appear to been seeing this issue too.

I am issuing a POST request that I expect to take around 15minutes to return a 200, I have set the timeout to 1200 in light of this.
The ansible task is only reported as executing once.
However in the logs of the endpoint receiving the request I see a second identical call hitting the endpoint almost exactly 5 minutes after the original request was made, I then see a third a further 7 minutes after that.
The initial call appears to run concurrently and is not terminated.
After the initial call succeeds I see no further repeated calls.

I will try and produce a test case and some tcpdump info.

**Comment by ansibot** [comment]
@lioramilbaum This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@lioramilbaum You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
